### PR TITLE
Support ResNeSt backbone

### PR DESCRIPTION
**Pull request by xvjiarui**

### CODE REVIEW

File: mmseg/models/backbones/resnest.py [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+        if self.radix > 1:
+            x = x.view(batch, self.groups, self.radix, -1).transpose(1, 2)
+            x = F.softmax(x, dim=1)
+            x = x.reshape(batch, -1)
+        else:
+            x = torch.sigmoid(x)
+        return x
+
+
+class SplAtConv2d(nn.Module):
```
Comment (hellock) [inline mmseg/models/backbones/resnest.py:38]: `SplitAttentionConv2d`

File: mmseg/models/backbones/resnest.py [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+from ..utils import ResLayer
+from .resnet import Bottleneck as _Bottleneck
+from .resnet import ResNetV1d
+
+
+class RSoftmax(nn.Module):
+    """Radix Softmax module in ``SplAtConv2d``.
+
+    Args:
+        radix (int): Radix of input. groups (int): Groups of input.
```
Comment (hellock) [inline mmseg/models/backbones/resnest.py:19]: Note the formating.

File: mmseg/models/backbones/resnest.py [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+
+    def __init__(self, radix, groups):
+        super().__init__()
+        self.radix = radix
+        self.groups = groups
+
+    def forward(self, x):
+        batch = x.size(0)
+        if self.radix > 1:
+            x = x.view(batch, self.groups, self.radix, -1).transpose(1, 2)
```
Comment (hellock) [inline mmseg/models/backbones/resnest.py:30]: Is the `transpose` necessary? Can we specify `dim=2` for softmax?

File: mmseg/models/backbones/resnest.py [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+
+    def forward(self, x):
+        x = self.conv(x)
+        x = self.norm0(x)
+        x = self.relu(x)
+
+        batch, rchannel = x.shape[:2]
+        if self.radix > 1:
+            splited = torch.split(x, rchannel // self.radix, dim=1)
+            gap = sum(splited)
```
Comment (hellock) [inline mmseg/models/backbones/resnest.py:124]: Use `torch.sum(splited, dim=0)` instead.

File: mmseg/models/backbones/resnest.py [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+        return getattr(self, self.norm1_name)
+
+    def forward(self, x):
+        x = self.conv(x)
+        x = self.norm0(x)
+        x = self.relu(x)
+
+        batch, rchannel = x.shape[:2]
+        if self.radix > 1:
+            splited = torch.split(x, rchannel // self.radix, dim=1)
```
Comment (hellock) [inline mmseg/models/backbones/resnest.py:123]: The passive voice of `split` is `split`, not `splited`.

File: mmseg/models/backbones/resnest.py (hunk 4 above)
Comment (xvjiarui) [reply mmseg/models/backbones/resnest.py:124]: `splited` is a list, `torch.sum` doesn't support.

File: mmseg/models/backbones/resnest.py (hunk 3 above)
Comment (xvjiarui) [reply mmseg/models/backbones/resnest.py:30]: `x` is reshaped to original shape afterward, which means the stack order of `self.group` and `self.radix` is changed (although `self.group=1` in current config).

File: mmseg/models/backbones/resnest.py (hunk 4 above)
Comment (hellock) [reply mmseg/models/backbones/resnest.py]:
```python
x = x.view(batch, rchannel // self.radix, -1, x.shape[2], x.shape[3])
gap = x.sum(dim=2)
```
may be faster?

Comment (xvjiarui) [reply mmseg/models/backbones/resnest.py]: Nice.
